packages/runtime/container-runtime/src/dataStoreContext.ts

```diff
@@ -1002,21 +1010,40 @@ export abstract class FluidDataStoreContext
 }
 
 export class RemoteFluidDataStoreContext extends FluidDataStoreContext {
-	private readonly initSnapshotValue: ISnapshotTree | undefined;
+	private initSnapshotValue: ISnapshotTree | undefined;
+	// Tells whether we need to fetch the snapshot before use. This is to support Data Virtualization.
+	private snapshotFetchRequired: boolean;
```

Comment:
I do not know if it's possible / simple to do but imagine that this layer always asks for snapshot from above. I.e. it calls into ContainerRuntime, provides (indirectly) it's path, it's groupID and leaves it up to Container Runtime to manage caching / fetching / etc. ContainerRuntime poses all the information to either satisfy this call  from cached data (in any form) or fetch from network

Reply:
Probably it's a separate item, but if it could be accomplished, it would be great.
Can we also get to a point where data store context does not keep snapshot forever? I.e. once it loaded from it, can it discard it / release memory? Our memory consumption is awful today - we have data in so many copies (though some of them are copy by reference).

Data store may need snapshot again and again (delay loading of DDSs), but I think it's Ok - it can ask for it container runtime many times. The key is that we are getting to a point where we have just one point of making decisions on how long to keep this info around. That said, such change is very risky, because that likely means container runtime can get rid of such snapshot, and next time you ask for it - it may give you snapshot form the future (loaded new instance), and DDSs loading code needs to deal with it (similar how you deal here by waiting, etc.), which is more work, so not sure we want to pick up that work / add risks.

So maybe this should be left for later time - not sure.

Reply:
Currently, Remote data store context does not call into container runtime before my change for snapshot. Container Runtime in passes that snapshot to it when creating a remote data store. But yes, I will check if we can avoid that and remote datastore always ask container runtime for the snapshot. 
If I find it complicated, then we can pursue that later and not touch the original logic i.e. without DV case.